libcoveoc4ids/additional_checks.py

```diff
@@ -0,0 +1,148 @@
+import gettext
+import re
+
+from libcoveoc4ids.additional_check import AdditionalCheck
+
+_ = gettext.gettext
+
+
+class CurrencyCheck(AdditionalCheck):
+    def process(self, data, flat_data):
+        amount_path = re.compile(".*amount$")
+
+        expected_currency_paths = []
+        missing = []
+
+        for path in flat_data.keys():
+            if amount_path.match(path):
+                expected_currency_paths.append(path[: -len("amount")] + "currency")
+
+        for expected in expected_currency_paths:
+            if expected not in flat_data:
+                missing.append(expected[: -len("currency")])
+
+        if len(missing) == 0:
+            return True
+
+        return self.result(
+            "missing-currency",
+            _(
+                "There are %(count)d values without a currency, Currencies should be published for all values"
```

Comment:
Minor grammar point. 

"There are %(count)d values without a currency**.** Currencies should be published for all values**.**"